src/haz3lcore/zipper/Printer.re

```diff
@@ -3,35 +3,39 @@ open Util.OptUtil.Syntax;
 
 let seg_of_zip = Zipper.seg_without_buffer;
 
-let rec of_segment = (~holes, seg: Segment.t): string =>
-  seg |> List.map(of_piece(~holes)) |> String.concat("")
-and of_piece = (~holes, p: Piece.t): string =>
+let rec of_segment = (~holes, ~concave_holes=None, seg: Segment.t): string =>
+  seg |> List.map(of_piece(~holes, ~concave_holes)) |> String.concat("")
+and of_piece = (~holes, ~concave_holes, p: Piece.t): string =>
   switch (p) {
-  | Tile(t) => of_tile(~holes, t)
+  | Tile(t) => of_tile(~holes, ~concave_holes, t)
+  | Grout({shape: Concave, _}) when concave_holes != None =>
+    Option.get(concave_holes)
   | Grout({shape: Concave, _}) => " "
   | Grout({shape: Convex, _}) when holes != None => Option.get(holes)
   | Grout({shape: Convex, _}) => " "
```

Comment:
Is there a reason we make these options as opposed to just defaulting `concave_hole`/`hole` to `" "`

Reply:
This question accidentally caused a refactor